### DISCUSSION TITLE
Cron Is Giving Me Issues

### DISCUSSION THREAD

**Discussion by forrestrose** [iptv-org/community#614] [Q&A]
First and foremost, thank you for the wealth of knowledge here and the tools created/given!

I'm a novice and not great at script but I can get the guide going and finding the channels no problem without cron.... but as soon as I add in a cron script, it lands at found 86 channels and then does nothing. Won't even let me throw in a new command. I'm not sure what I'm doing wrong. Here's what I'm seeing:

PS C:\Users\forre\epg> npm run grab -- --site=tvguide.com --maxConnections=10 --cron "0 */6 * * *" run serve

> grab
> npx tsx scripts/commands/epg/grab.ts --site=tvguide.com --maxConnections=10 --cron 0 */6 * * * run serve

(node:440352) ExperimentalWarning: `--experimental-loader` may be removed in the future; instead use `register()`:
--import 'data:text/javascript,import { register } from "node:module"; import { pathToFileURL } from "node:url"; register("file%3A///C%3A/Users/forre/epg/node_modules/tsx/dist/loader.mjs", pathToFileURL("./"));'
(Use `node --trace-warnings ...` to show where the warning was created)
(node:440352) UnsupportedWarning: `globalPreload` has been removed; use `initialize` instead.
(Use `node --trace-warnings ...` to show where the warning was created)
starting...
config:
  output: guide.xml
  maxConnections: 10
  gzip: false
  site: tvguide.com
  cron: 0 */6 * * *
loading channels...
(node:440352) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
  found 86 channel(s)
 

Thanks in advance!

**Comment by forrestrose** [comment]
I guess I just have to leave powershell open and it runs at that given time--as long as powershell is open.

**Comment by DimSumDev** [comment]
I'm not an expert and just started using the EPG repo scripts.

Seems the cron on `npm run grab` vs a cron job on a server are not the same. The cron on npm run waits to run at a specified time and then finishes.

If you want to run this on a scheduled time, create a cron job on your OS system. 
If you only want to use the `npm run cron` , then you need to use a detached session, so your application can run in background and your screen is not waiting. Linux uses "screen" or "tmux"
 
My suggestion would be to get an old cheap computer to turn into a local server to grab and serve.
Use any free Linux flavor.

My set-up is a pi 3 w/ Ubuntu to grab and serve locally. (this is testing before I move it permanently to home server or cloud server)
If you want to spend some more money, get a small cheap used "nuc"  (about 150) Linux server as a local media server and what other "applications" you want to run.
